coder-sdk/workspace.go

```diff
@@ -364,3 +366,99 @@ func (c *DefaultClient) WorkspacesByWorkspaceProvider(ctx context.Context, wpID
 	}
 	return workspaces, nil
 }
+
+const (
+	// SkipTemplateOrg allows skipping checks on organizations.
+	SkipTemplateOrg = "SKIP_ORG"
+)
+
+type TemplateScope string
+
+const (
+	// TemplateScopeSite is the scope for a site wide policy template.
+	TemplateScopeSite = "site"
+)
+
+type SetPolicyTemplateRequest struct {
+	TemplateID string `json:"template_id"`
+	Type       string `json:"type"` // site, org
+}
+
+type SetPolicyTemplateResponse struct {
+	MergeConflicts []*WorkspaceTemplateMergeConflict `json:"merge_conflicts"`
+}
+
+type WorkspaceTemplateMergeConflict struct {
+	WorkspaceID             string    `json:"workspace_id"`
+	CurrentTemplateWarnings []string  `json:"current_template_warnings"`
+	CurrentTemplateError    *TplError `json:"current_template_errors"`
+	LatestTemplateWarnings  []string  `json:"latest_template_warnings"`
+	LatestTemplateError     *TplError `json:"latest_template_errors"`
+	CurrentTemplateIsLatest bool      `json:"current_template_is_latest"`
+	Message                 string    `json:"message"`
+}
+
+func (mc WorkspaceTemplateMergeConflict) String() string {
+	var sb strings.Builder
+
+	if mc.Message != "" {
+		sb.WriteString(mc.Message)
+	}
+
+	currentConflicts := len(mc.CurrentTemplateWarnings) != 0 || mc.CurrentTemplateError != nil
+	updateConflicts := len(mc.LatestTemplateWarnings) != 0 || mc.LatestTemplateError != nil
+
+	if !currentConflicts && !updateConflicts {
+		sb.WriteString("No conflicts\n")
+		return sb.String()
+	}
+
+	if currentConflicts {
+		if len(mc.CurrentTemplateWarnings) != 0 {
+			sb.WriteString(fmt.Sprintf("Warnings: \n%s\n", strings.Join(mc.CurrentTemplateWarnings, "\n")))
+		}
+		if mc.CurrentTemplateError != nil {
+			sb.WriteString(fmt.Sprintf("Errors: \n%s\n", strings.Join(mc.CurrentTemplateError.Msgs, "\n")))
+		}
+	}
+
+	if !mc.CurrentTemplateIsLatest && updateConflicts {
+		sb.WriteString("If workspace is updated to the latest template:\n")
+		if len(mc.LatestTemplateWarnings) != 0 {
+			sb.WriteString(fmt.Sprintf("Warnings: \n%s\n", strings.Join(mc.LatestTemplateWarnings, "\n")))
+		}
+		if mc.LatestTemplateError != nil {
+			sb.WriteString(fmt.Sprintf("Errors: \n%s\n", strings.Join(mc.LatestTemplateError.Msgs, "\n")))
+		}
+	}
+
+	return sb.String()
```

Comment:
Something that might be helpful is some sort of summary at the end. I have 200 workspaces in my dogfood, and it would be nice to see something like:

```
102 Workspaces will not be able to be rebuilt.
152 Workspaces have warnings but are still functional.
```

In the future, it would be great to merge-common errors and report which users are affected maybe? But for now I think a simple summary like that would be great.

Reply:
I added a summary at the end.

Mine isn't too exciting bc I regularly delete my workspaces in my dev deployment, so I only have like 2.
I'm curious to see what you think of the summary when there are 200 🤔

Reply:
I'm very curious about why @Emyrk 200 workspaces :joy:

Reply:
Ha, my deployment has 200 workspaces because I was testing the response times of our apis at those numbers. I made a little deployment branch to make them fast, but I never made anything to delete them in bulk haha.

I figure I'll just keep em as a bit of a "stress test" for the UI

Reply:
The summary at 200 is not the most helpful thing, but I'm not worried about optimizing for 200 workspaces right now. Just thought a summary would be nice even for like ~30 workspaces which is reasonable at a company